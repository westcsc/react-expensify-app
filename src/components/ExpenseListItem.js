import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => {

    console.log(id, description, amount,createdAt);

    return (
        <div>
            <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
            </Link>
                
        <p>{amount} - { moment(createdAt).format('MMM Do, YYYY') }</p>
        </div>
    )
}
 
export default ExpenseListItem;