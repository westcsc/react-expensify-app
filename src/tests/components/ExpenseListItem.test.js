import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render Expense List Item with expense data', () => {
    const expense = expenses[1];
    const wrapper = shallow(<ExpenseListItem {...expense}/>);
    expect(wrapper).toMatchSnapshot();
});