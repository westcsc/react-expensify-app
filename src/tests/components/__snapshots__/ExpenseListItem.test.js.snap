// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense List Item with expense data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    109500
     - 
    Dec 27th, 1969
  </p>
</div>
`;
